app/controllers/map_controller.rb

```diff
@@ -16,7 +16,10 @@ def index
   end
 
   def show
-    @node = DrupalNode.find_map_by_slug(params[:name]+'/'+params[:date])
+    @node = DrupalNode.find params[:name]
```

Comment:
Uh ok I think there is a bit misunderstanding in this part. The `params[:name]` basically contains the slug of the map not just the name. So the query searches for the map based on the friendly_id slug. So if there is a map with a similar name but a different date it will go as `/map/NewYork-map--2/07-22-2016`. So you see there are no two slugs which are same.